volume-group-operator/pkg/client/client.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/kubernetes-csi/csi-lib-utils/connection"
+	"github.com/kubernetes-csi/csi-lib-utils/metrics"
+	"github.com/kubernetes-csi/csi-lib-utils/rpc"
+	"google.golang.org/grpc"
+)
+
+// Client holds the GRPC connenction details.
+type Client struct {
+	Client  *grpc.ClientConn
+	Timeout time.Duration
+}
+
+// Connect to the GRPC client.
```

Comment:
do we need to add comments above functions? we don't do it that much in other places of code in our product

Reply:
this id doc, not comment, and community doing it..

Reply:
I what is the difference between this comment and another comment that I will put above the function in the controller? IMO we should keep the same standard in all of our repositories regarding the repository purpose

Reply:
It's not a comment, it's doc and common practice in community code.
its equal to the doc in the interface in the controller
if we want to put it in the community we should keep their convention

Reply:
[this](https://github.com/IBM/ibm-block-csi-operator/blob/3559d9654df02fae3325520f74380a4e18a981aa/volume-group-operator/pkg/client/volumegroup_client.go#L48) is also considered a doc?

Reply:
and regarding the community, we didn't contribute this code to the community yet, so I think we should act under our formatting, if we will want to contribute it to the community and if they will require it, we can add it, but if those comments are important to you so we can leave them like this, just wanted to understand what is the difference between the functions in this PR than functions in other places in our code and if I need to add comments as well on the functions that I will write to the volumeGroup operator

Reply:
> [this](https://github.com/IBM/ibm-block-csi-operator/blob/3559d9654df02fae3325520f74380a4e18a981aa/volume-group-operator/pkg/client/volumegroup_client.go#L48) is also considered a doc?

yes

Reply:
this is go formatting.

Reply:
if it's go formatting then do we need to open a ticket to change all the functions in the node that will have a doc above it?
and this way of go is the equivalent of python docstring?

Reply:
[done](https://github.com/IBM/ibm-block-csi-operator/pull/294/commits/f2664e32ef1512cd27e826b036f6a1d70fb518d7)

Reply:
thanks